multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -87,7 +88,16 @@
             }
 
             try (InputStream in = item.openStream()) {
-                FileEntry entry = fileService.addFile(spaceGuid, namespace, item.getName(), in);
+                FileEntry entry;
+                String fileSizeHeader = request.getHeader("X-File-Size");
```

Comment:
You could extract this and the following 3 lines of code in a separate method called `getFileSizeHeaderValue` and let it return a Long instead of a String.